python/paddle/fluid/optimizer.py

```diff
@@ -35,7 +36,11 @@ class Optimizer(object):
     but need to use one of it's implementation.
     """
 
-    def __init__(self, learning_rate, regularization=None):
+    def __init__(self,
+                 learning_rate,
+                 global_step=None,
+                 regularization=None,
+                 append_all_reduce=False):
```

Comment:
I feel this to be an implementation detail. Can we avoid exposing it to user?

Reply:
Sure.